mathlibs/src/CMakeLists.txt

```diff
@@ -6,7 +6,9 @@ IF(NOT LAPACK_FOUND)
   ADD_SUBDIRECTORY(lapack)
 ENDIF()
 
-ADD_SUBDIRECTORY(arpack)
-IF(MPI_FOUND)
-  ADD_SUBDIRECTORY(parpack)
+IF(NOT EXTERNAL_ARPACK)
+  ADD_SUBDIRECTORY(arpack)
+  IF(MPI_FOUND)
+    ADD_SUBDIRECTORY(parpack)
+  ENDIF()
```

Comment:
Should it be possible to use an external ARPACK in combination with the PARPACK that is bundled with Elmer?
What do you think about this change?
``` suggestion
IF(NOT ARPACK_FOUND)
  ADD_SUBDIRECTORY(arpack)
ENDIF()
IF(MPI_FOUND AND NOT PARPACK_FOUND)
  ADD_SUBDIRECTORY(parpack)
ENDIF()
```

Reply:
Yes, depending on the components' search results, you can have a mixed configuration (arpack: external and parpack: internal, or vice-versa). Your proposed change would act as a failsafe mechanism if something went wrong. However, I think it could defeat the original idea of explicitly using external libraries. 
Plus, some platforms provide separate packages for arpack and parpack. So if the user forgets to install both, this omission could go unnoticed if they ignore the cmake's output.

Perhaps the solution to have a mixed configuration is to split the logic of FindARPACK.cmake into two scripts (FindARPACK.cmake and FindPARPACK.cmake) and add a new variable (EXTERNAL_PARPACK::BOOL). 
What do you think?